model-optimizer/mo/main.py

```diff
@@ -144,20 +144,18 @@ def prepare_ir(argv: argparse.Namespace):
 
     # This try-except is additional reinsurance that the IE
     # dependency search does not break the MO pipeline
-    try:
-        argv.ie_is_available = find_ie_version(silent=argv.silent)
-
-        if not argv.ie_is_available and not argv.silent:
-            print("[ WARNING ] Could not find the Inference Engine Python API. At this moment, the Inference Engine dependency is not required, but will be required in future releases.")
-            print("[ WARNING ] Consider building the Inference Engine Python API from sources or try to install OpenVINO (TM) Toolkit using \"install_prerequisites.{}\"".format(
+    def raise_ie_not_found():
+        raise Error("Could not find the Inference Engine or nGraph Python API.\n"
+                    "Consider building the Inference Engine and nGraph Python APIs from sources or try to install OpenVINO (TM) Toolkit using \"install_prerequisites.{}\"".format(
                     "bat" if sys.platform == "windows" else "sh"))
-            # If the IE was not found, it will not print the MO version, so we have to print it manually
-            print("{}: \t{}".format("Model Optimizer version", get_version()))
+    try:
+        if not find_ie_version(silent=argv.silent):
+            raise_ie_not_found()
     except Exception as e:
-        argv.ie_is_available = False
+        raise_ie_not_found()
```

Comment:
Should we skip this check if we enable legacy IR generation with a dedicated command line option? 
Or we should want to remove the legacy IR generation possibility then we need to remove command line parameter and the rest of the code.

Reply:
The second option :)
Legacy IR generation will be removed.